### PR TITLE
Change the package name to phplist/core

### PR DESCRIPTION
**Pull request by samtuke**
### Summary

Changed composer.json to reflect the updated repository name following discussion here in://github.com/phpList/core/issues/273

### CODE REVIEW

**Comment by oliverklee** [comment]
The commit message is misleading:

You're not changing the repository name (that would be a setting in the GitHub repository and an update of the package setting in Packagist), you're changing the package name.

And it should generally be "[This commit will] Change …",  not "[This commit will] Changed …".

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 ### Changed
 - Remove the obsolete core classes (#267)
 - Adopt more of the default Symfony project structure (#265, #268, #269, #270)
+- Change composer configuration to reflect new repository name 'core'
```
Comment (oliverklee) [inline CHANGELOG.md]: This needs to go to the top of the changes list (as it's ordered "newest first").

**Comment by samtuke** [comment]
@oliverklee Thanks -- fixed changelog order and commit message

**Comment by oliverklee** [comment]
You'll also need to update the badges and installation instructions in the README.

**Comment by oliverklee** [comment]
We'll be able to change the package name on Packagist after this has been merged. We'll then need to update the dependencies in the other packages, and (as a separate step after that) change the PHP namespaces.

**Comment by oliverklee** [comment]
And the version number "4" from the README should also go away.

**Comment by samtuke** [comment]
@oliverklee Badge references updated

**Comment by samtuke** [comment]
@oliverklee Version number 4 references removed in README.md

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 - Bidirectional m:n association Subscribers/SubscriberLists (#254)
 
 ### Changed
+- Change composer configuration to reflect new repository name 'core'
```
Comment (oliverklee) [inline CHANGELOG.md]: I propose this text instead:

Rename the Composer package to "phplist/core"

Please also add the PR number.

File: src/Composer/ScriptHandler.php [hunk 3]
```diff
@@ -19,7 +19,7 @@ class ScriptHandler extends SensioScriptHandler
     /**
      * @var string
      */
-    const CORE_PACKAGE_NAME = 'phplist/phplist4-core';
+    const CORE_PACKAGE_NAME = 'phplist/core';
```
Comment (oliverklee) [inline src/Composer/ScriptHandler.php]: This constant should move to the module finder, and the module finder should use it as well.

**Pull request merged**